src/Microsoft.DotNet.Cli.Utils/StreamForwarder.cs

```diff
@@ -83,11 +83,6 @@ public void Read(TextReader reader)
 
         private void WriteBuilder()
         {
-            if (_builder.Length == 0)
```

Comment:
@eerhardt The case you mentioned is handled by ignoring `\r` characters altogether in the input, and relying on WriteLine to insert the appropriate newline for the platform.

See: `! s_ignoreCharacters.Contains(currentCharacter)'

Your proposed change would end up omitting the `\r` altogether (since none is appended in the WriteBuilder function).

I don't think whether or not to include the newlines in the builder is the heart of the issue here though. Since we are doing a WriteLine per newline, the behavior should be identical. 

The heart of the issue is what do we do when we encounter `\n\n` or `\r\n\r\n` as an input? Currently we'd do two empty writelines, which I recently discovered won't write anything (not even a newline). 

Including the newlines in the builder is one solution to this problem (which your suggested change @eerhardt would fix). 

I think the ideal fix here is to stop special casing for `\r` and go with eric's suggestion, which would look like this:

```
            while ((readCharacterCount = reader.Read(buffer, 0, bufferSize)) > 0)
             {
                 currentCharacter = buffer[0];

                 _builder.Append(currentCharacter);

                    if (currentCharacter == s_flushBuilderCharacter)
                    {
                           WriteBuilder();
                     }
                 
             }
```